lib/srv/reexec.go

```diff
@@ -93,6 +94,31 @@ type ExecCommand struct {
 
 	// IsTestStub is used by tests to mock the shell.
 	IsTestStub bool `json:"is_test_stub"`
+
+	// RemoteAddr is the address of the remote host.
+	RemoteAddr string `json:"remote_addr"`
+}
+
+func createUaccSession(ttyName string, c *ExecCommand) (bool, error) {
+	remoteStringIP, _, _ := net.SplitHostPort(c.RemoteAddr)
+	remoteIP := net.ParseIP(remoteStringIP)
+	hostname, err := os.Hostname()
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	err = uacc.Open(c.Login, hostname, remoteIP, ttyName)
+	if err != nil {
+		return true, trace.Wrap(err)
```

Comment:
Hmm, I missed it on the previous run - what does the boolean return stand here for so it's clear why `true` means error?

Reply:
The boolean here is if it's a "soft" error that we should just gracefully log and move on or if something is gravely wrong and the caller should return and propagate it upwards.

Reply:
Added docs.

Reply:
I would not bother adding another layer of complexity. If the error can be ignored for a good reason - let's just log it and return an error for "gravely" cases.